feat: [1/n] Teams Downgrade Flow Experiment Setup

surbhi-posthog opened the pull request:
## Problem

When a user unsubscribes from the teams feature, we want to show them a list of features that they are currently using and will lose access to.

## Changes

In this PR, I'm doing the initial experiment setup and showing the user a basic modal before they can unsubscribe. Next I will add logic to check which features are actually used by the org and update the list.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Cloud: Yes

## How did you test this code?

https://github.com/user-attachments/assets/c7d906e4-9d78-468c-891c-9d2706c68486


Verified modal pops up before they are allowed to unbscribed.
Verified that the user only sees the new flow if they are in the experiment group

surbhi-posthog committed:
experiement setup

surbhi-posthog committed:
adding the modal to the globals

zlwaterfield reviewed (commented):
This is a great start - I wonder if we should be adding this `UnsubscribeSurveyModal`  to the existing as step 1. Few reasons
1. TeamsDowngradeModal is too specific to the teams plan and it should be able to be more generic 
2. I don't like putting things in Global Modals unless they really need it and this is only on the billing page so shouldn't need to be. 
3. If then continue to cancel we want it to be a seamless flow to the next unsubscribe step.

surbhi-posthog commented:
Thanks for the feedback. A few comments in response:

> 1. TeamsDowngradeModal is too specific to the teams plan and it should be able to be more generic
I see, I guess you are thinking we can re-use this for any unsubscribe event right, like for example for enterprise plan? In that case, i can build something a bit more generic.

> 2. I don't like putting things in Global Modals unless they really need it and this is only on the billing page so shouldn't need to be.
Okie, I can find another spot for it.

> 3. If then continue to cancel we want it to be a seamless flow to the next unsubscribe step.
Please review the test plan. I just added a video of what the flow looks like. If the question is to show the survey before the feature list, I'm ok with that but we can add always make changes once the feature is complete since that change should be fairly simple.

surbhi-posthog committed:
comments

surbhi-posthog commented:
Addressing comments above: 

1. Move the modal outside the `GlobalModals.tsx` and render it in` BillingProductAddonActions.tsx`
2. Added a `FeatureDowngradeModal.tsx` to make the modal generic
3. Also updated the test plan with a video of the current flow

If we need to make changes on the unsubscribe flow - ie. show the survey before the modal, I can make those changes upstack

surbhi-posthog commented:
I'm moving this to a new branch since that was easier: https://github.com/PostHog/posthog/pull/27617
